docs/dapp/sapphire/browser.md

```diff
@@ -0,0 +1,231 @@
+---
+description: Writing Sapphire dApp for browser and Metamask
+---
+
+# Browser Support
+
+Confidential Sapphire dApps work in web browsers by wrapping the
+Metamask extension for signing and encrypting transactions.
+
+Let's begin with the [Hardhat boilerplate]. As mentioned on their website
+the boilerplate provides the following:
+
+- The Solidity contract implementing an ERC-20 token
+- Tests for the entire functionality of the contract
+- A minimal React front-end to interact with the contract using ethers.js
+
+Go ahead and clone the original [Hardhat boilerplate repo]. Move to the checked
+out folder and apply the Sapphire-specific changes to the Hardhat config file
+as described in the [quickstart].
+
+Install dependencies and deploy the contract on the Testnet:
+
+```
+pnpm install
+pnpm add -D @oasisprotocol/sapphire-paratime
+PRIVATE_KEY="0x..." pnpm hardhat run scripts/deploy.js --network sapphire
+```
+
+This will compile the contract and deploy it on the Testnet. In addition
+to the quickstart steps, **the contract address and ABI will also automatically
+be copied over to the `frontend/src/contracts` folder so that the frontend can
+access them!**
+
+:::warning
+
+The contract in the Hardhat boilerplate is ERC-20-compatible and emits the
+`transfer` event. If your wish to preserve confidentiality, you can comment
+out [line 66]. Read [the guide](guide.mdx#contract-logs) to learn more.
+
+:::
+
+## Frontend
+
+Move into the `frontend` folder and install dependencies:
+
+```shell
+pnpm install
+pnpm add -D @oasisprotocol/sapphire-paratime
+```
+
+### `http` caveat
+
+The `@oasisprotocol/sapphire-paratime` npm package for wrapping Web3 objects
+uses `http` internally for validating the Sapphire endpoint certificates. Since
+the browser supports validating certificates natively, this package is only
+required in the Node.js environment. Unfortunately, there is no simple way to
+convince Webpack in React not to bundle `http` along when compiling the app for
+the browser ([learn more][React Webpack issue]).
+
+To overcome this, do the following:
+
+1. Add `react-app-rewired` dependency:
+
+  ```shell
+  pnpm add -D react-app-rewired
+  ```
+   
+2. Configure it by creating `config-overrides.js` with the following content:
+
+  ```js title="frontend/config-overrides.js"
+  module.exports = function override(config, env) {
+    okv = {
+      fallback: {
+        https: false,
+      },
+    };
+  
+    if (!config.resolve) {
+      config.resolve = okv;
+    } else {
+      config.resolve.fallback = { ...config.resolve.fallback, ...okv.fallback };
```

Comment:
This looks weird to me because of the three dots, but I don't do js, so it may be ok?